### PR TITLE
Popup Component and dlgEnter Implementation

### LINKED ISSUE #5596: Add examples to the Prepare > Generate > Enter dialogue
@Patowhiz here is the current Enter dialogue.

I am interested in adding some examples.  If so, then I wonder how this should be done?  
We can have tooltips, of course.
But in addition I wonder about helping users in the writing of examples of the commands as we do in the File > New Data Frame dialogue.

Here there are 2 obvious ways.  We could have a button - as we do in the File > New Data Frame.  I assume that would be quick and easy to do, following the File >New Data Frame work.  It would be just the right-hand side of the formula.  It could go just after the show arguments checkbox.

Or the dialogue automatically remembers past commands.  When you start it is empty.  It could start with a few examples - perhaps below a horizontal line.  

@Patowhiz What do you think?  Should we add the examples?  Would it be very easy with a button?  If so, then I am slightly inclined that way.

I got to this because I would like to build towards a guide to help beginners on their first commands that use these "halfway" dialogues.

### LINKED ISSUE #5876: Popup Component
Currently we have code that is responsible for creating a custom "popup" in different dialogs and controls. As discussed with @dannyparsons there's need to have this code in a central place which will help in future maintenance, enhancement and avoid repetition of code as more dialogs continue taking advantage of this feature. 

Check screenshots of the popups below
popup1

popup2

### PR DESCRIPTION
**Pull request by Patowhiz**
Fixes #5596 . Fixes #5876.

@africanmathsinitiative/developers this is ready for review.

This issue implements the popup component and also demonstrates through dlgEnter how other dialogs can use it. It's a continuation of PR #5877(which should be merged first) which has an empty class that represents the popup component. 

Once merged, the other dialogs that currently have code that is implemented in this component, can be amended to use it. 

@rdstern as discussed in issue #5596 the only thing waiting in the dialog are the samples and their tooltips. I will add them as soon as I get your feedback.

### CODE REVIEW

**Comment by lloyddewit** [comment]
@Ivanluv Please could you review this? thanks

**Comment by lloyddewit** [comment]
@Ivanluv Did you get a chance to look at this? thanks

**Comment by Ivanluv** [comment]
@lloyddewit  am looking at this now

**Comment by Ivanluv** [comment]
@lloyddewit  the code looks fine from my review

**Comment by lloyddewit** [comment]
@Ivanluv Thank you for the peer review!

@rdstern Above, @Patowhiz  requested '@rdstern as discussed in issue #5596 the only thing waiting in the dialog are the samples and their tooltips. I will add them as soon as I get your feedback.`. If you are able to provide these then I can do the final review and merge, thanks

**Comment by Patowhiz** [comment]
@rdstern 

> However, the examples that are given do not move into the Data field

 This is is now fixed, was a bug in checking for selection.

> which incidentally should be better lined up and also have a colon
I don't understand what you mean by should have colon and lined, in the popup? Below is the image of the current design during runtime.

dlgEnterNew

**Comment by rdstern** [comment]
Just that the label "Data" should move up slightly and become "Data:"

**Comment by Patowhiz** [comment]
@rdstern that's now fixed. You can test the changes and @lloyddewit can do the review

**Comment by lloyddewit** [comment]
@rdstern Please could you review? thanks

File: instat/clsPopup.vb [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+' (at your option) any later version.
+'
+' This program is distributed in the hope that it will be useful,
+' but WITHOUT ANY WARRANTY; without even the implied warranty of
+' MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+' GNU General Public License for more details.
+'
+' You should have received a copy of the GNU General Public License 
+' along with this program.  If not, see <http://www.gnu.org/licenses/>.
+Public Class clsPopup
```
[the remark this answers is not in the document]
Comment (Patowhiz) [reply instat/clsPopup.vb:36]: done

File: instat/clsPopup.vb [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    'event raised when the contents control loses focus. Indeicates that the popup is closing
```
[the remark this answers is not in the document]
Comment (Patowhiz) [reply instat/clsPopup.vb]: done

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern Only the comments changed since your previous approval. Please could you approve again? thanks

**Review by rdstern — APPROVED** [review]
I can't see any difference.  This will still be completed by @Wycklife later

**Pull request merged**